File: metodos/secante.py
def secante(f, x0, x1, tol, max_iter=100):
    fx0 = f(x0)
    fx1 = f(x1)
    i = 0
    while abs(fx1) > tol and i < max_iter:
        try:
            dx = (x1 - x0)/(fx1 - fx0)
            x0 = x1
            fx0 = fx1
            x1 = x1 - fx1*dx
            fx1 = f(x1)
            i += 1
        except ZeroDivisionError:
            print("Error: División por cero.")
            return None
    if abs(fx1) > tol:
        print("El método no converge después de %d iteraciones." % max_iter)
        return None
    return x1

# F(x)
def f(x):
    return x**2 - 2

# intervalo y toleracia
x0 = 1
x1 = 2
tol = 1e-6

raiz = secante(f, x0, x1, tol)
print("La raíz de la función es:", raiz)

File: metodos/MetodosJava/Secante.java
import java.util.function.Function;

public class Secante {
    public static String secante(Function<Double, Double> f, double x0, double x1, double tol, int maxIter) {
        int iter = 0;
        double fx0 = f.apply(x0);
        double fx1 = f.apply(x1);

        while (iter < maxIter) {
            try {
                double dx = (x1 - x0) / (fx1 - fx0);
                x0 = x1;
                fx0 = fx1;
                x1 = x1 - fx1 * dx;
                fx1 = f.apply(x1);

                if (Math.abs(fx1) <= tol) {
                    return "La raíz de la función es: " + x1;
                }

                iter++;
            } catch (ArithmeticException e) {
                return "Error: División por cero.";
            }
        }

        return "El método no converge después de " + maxIter + " iteraciones.";
    }
}
